zyte_spider_templates/spiders/base.py

```diff
@@ -81,6 +121,65 @@ class BaseSpiderParams(BaseModel):
         },
     )
 
+    @field_validator("urls", mode="before")
+    @classmethod
+    def validate_url_list(cls, value: Union[List[str], str]) -> List[str]:
+        """Validate a list of URLs.
+
+        If a string is received as input, it is split into multiple strings
+        on new lines.
+
+        List items that do not match a URL pattern trigger a warning and are
+        removed from the list. If all URLs are invalid, validation fails.
+        """
+        if isinstance(value, str):
+            value = value.split("\n")
+        if value:
```

Comment:
Any of the actually supported types: `None` or a list of `str`.

Reply:
Probably in this case we also need to validate urls inside the list.

Reply:
@Gallaecio will you add something to make this work?
[mind that "URL\nURL" will not work](https://stackoverflow.com/q/3005963/939364) - from your comment in description.
or running like
`scrapy crawl spider -a urls="https://www.nytimes.com/\nhttps://www.theguardian.com"`
I've added this workaround https://github.com/zytedata/zyte-spider-templates-private/blob/article/zyte_spider_templates/spiders/article.py#L135, but might be we can handle it in `validate_url_list` too.

Reply:
I think supporting that syntax is a bit dangerous, as `\n` are valid URL characters. Since it will not be an issue in the UI or from code, I am not sure if it is worth supporting.

Maybe we could support spaces as a separator, but we risk scenarios were users expect a space → %20 conversion to happen.

Reply:
@Gallaecio I've checked https://www.rfc-editor.org/rfc/rfc3986 and haven't found anything about \n as allowed character - but might be missed something.

Reply:
I meant valid character**s**, i.e. `\` is a valid character and `n` is a valid character. https://example.com/\n is a valid URL.

Reply:
Got it, makes sense, thanks!